command/sub8_mission_control/src/goal_client.py

```diff
@@ -0,0 +1,152 @@
+#! /usr/bin/env python
+
+import rospy
+import actionlib
+import roslib; roslib.load_manifest('sub8_mission_control')
+import sub8_mission_control.msg
+from nav_msgs.msg import Odometry
+from sub8_ros_tools import geometry_helpers as gh
+
+class goal_proxy(object):
+    
+    '''
+        This class is used to send waypoint goals to the motion planner
+        It contains several shortcuts to send movements as well as the 
+        ability to specifif location information
+    '''
+
+    def __init__(self, testing):
+        
+        self.current_pos = Odometry()
+        self.client = actionlib.SimpleActionClient('goal', sub8_mission_control.msg.current_goalAction)
+        rospy.Subscriber('/odom', Odometry, self.odom_cb)
```

Comment:
Have you tried running this? I believe this subscriber might get garbage collected